app/controllers/admin/videos_controller.rb

```diff
@@ -0,0 +1,31 @@
+class Admin::VideosController < ApplicationController 
+  before_action :require_user
+  before_action :require_admin
+  
+  def new
+    @video = Video.new
+  end
+  
+  def create
+    @video = Video.create(video_params)
+    if @video.save
+      flash['success'] = "#{@video.title} has been added."
+      redirect_to new_admin_video_path
+    else
+      flash['danger'] = "Video has not been added."
```

Comment:
this should be `flash.now['danger']` you want the flash *now* when you render but his flash is fast enough with a redirect.